adapters/mqtt-vertx/src/test/java/org/eclipse/hono/adapter/mqtt/impl/VertxBasedMqttProtocolAdapterTest.java

```diff
@@ -257,37 +306,58 @@ public void testMapTopicSupportsShortAndLongTopicNames(final VertxTestContext ct
 
     }
 
-    private void givenAnAdapter() {
-
-        config = new MqttProtocolAdapterProperties();
-        adapter = new VertxBasedMqttProtocolAdapter();
-        adapter.setConfig(config);
-    }
-
-    private static MqttPublishMessage newMessage(final MqttQoS qosLevel, final String topic) {
-        return newMessage(qosLevel, topic, Buffer.buffer("test"));
-    }
+    /**
+     * Verifies that the adapter calls the mapper if configured. Also the updated payload and deviceId is checked.
+     *
+     * @param ctx The helper to use for running tests on vert.x.
+     */
+    @Test
+    public void testMapperShouldUpdatePayloadAndDeviceId(final VertxTestContext ctx) {
+        givenAnAdapterWithMapper();
+        final MessageMapping messageMapping = new MessageMapping(mapperWebClient, config);
+
+        final ResourceIdentifier targetAddress = ResourceIdentifier.from(TelemetryConstants.TELEMETRY_ENDPOINT, TEST_TENANT_ID, TEST_DEVICE);
+        final HttpRequest httpRequest = mock(HttpRequest.class);
+        when(mapperWebClient.post(any(int.class), any(String.class), any(String.class))).thenReturn(httpRequest);
+        when(httpRequest.putHeaders(any(MultiMap.class))).thenReturn(httpRequest);
+        when(httpRequest.ssl(any(Boolean.class))).thenReturn(httpRequest);
+        final AsyncResult asyncResult = mock(AsyncResult.class);
+        when(asyncResult.succeeded()).thenReturn(true);
+        final HttpResponse httpResponse = mock(HttpResponse.class);
+        when(asyncResult.result()).thenReturn(httpResponse);
+        final VertxHttpHeaders headers = new VertxHttpHeaders();
+        final String newDeviceId = "new-device";
+        headers.add(MessageHelper.APP_PROPERTY_DEVICE_ID, newDeviceId);
+        when(httpResponse.headers()).thenReturn(headers);
+        final Buffer changedBuffer = Buffer.buffer("changed");
+        when(httpResponse.bodyAsBuffer()).thenReturn(changedBuffer);
+
+        final MqttPublishMessage message = newMessage(MqttQoS.AT_LEAST_ONCE, EventConstants.EVENT_ENDPOINT);
+        final MqttContext context = newContext(message, null);
+        final MapperEndpoint mapperEndpoint = MapperEndpoint.from("host", 1234, "/uri", false);
+        messageMapping.mapMessageRequest(context, targetAddress, message, mapperEndpoint, headers).onComplete(ctx.succeeding(mappedMessage -> {
+        }));
```

Comment:
This actually seems to be a test for the `MessageMapping` class but not for the protocol adapter, doesn't it?
Also, it doesn't seem to test anything as there are no assertions, or am I mistaken?